src/main/java/appeng/core/sync/packets/PacketMEFluidInventoryUpdate.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2018, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.core.sync.packets;
+
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.BufferOverflowException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import javax.annotation.Nullable;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraftforge.fml.common.network.internal.FMLProxyPacket;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.client.gui.implementations.GuiFluidTerminal;
+import appeng.core.AELog;
+import appeng.core.sync.AppEngPacket;
+import appeng.core.sync.network.INetworkInfo;
+import appeng.util.item.AEFluidStack;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 22/05/2018
+ * @since rv6 22/05/2018
+ */
+public class PacketMEFluidInventoryUpdate extends AppEngPacket
+{
+	private static final int UNCOMPRESSED_PACKET_BYTE_LIMIT = 16 * 1024 * 1024;
+	private static final int OPERATION_BYTE_LIMIT = 2 * 1024;
+	private static final int TEMP_BUFFER_SIZE = 1024;
+	private static final int STREAM_MASK = 0xff;
+
+	// input.
+	@Nullable
+	private final List<IAEFluidStack> list;
+	// output...
+	private final byte ref;
+
+	@Nullable
+	private final ByteBuf data;
+	@Nullable
+	private final GZIPOutputStream compressFrame;
+
+	private int writtenBytes = 0;
+	private boolean empty = true;
+
+	// automatic.
+	public PacketMEFluidInventoryUpdate( final ByteBuf stream ) throws IOException
+	{
+		this.data = null;
+		this.compressFrame = null;
+		this.list = new LinkedList<>();
+		this.ref = stream.readByte();
+
+		// int originalBytes = stream.readableBytes();
+
+		final GZIPInputStream gzReader = new GZIPInputStream( new InputStream()
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "GZIPInputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2095)